### ISSUE TITLE
Feature request: label creation and attachment

### ISSUE THREAD

**Issue by mgrojo** [OpenNingia/vscode-clearcase#168]
Hi,

Thanks for this extension, it is working better for me than the official IBM version because I cannot use CCRC.

One important feature of ClearCase that I miss is creation of label types and attachment to versions. Specifically, to files just checked-in.

A workflow that I like is creating a label type with an appropriate comment for the task, and then (I was doing this with the help of a script) attaching the label to every checkout done in the view and assigning that comment to the checkout. You could implement it with new operation to create the label type with a comment. Then in the checkout operation a prompt for an optional label, and you checkout the file with the label comment and attach the label. In the check-in operation, the comment is left as is if the user leaves the comment field empty and also leave the labels of files that already have it.

Another option, which might be easier to implement, is to add another field to the "check-in" dialog for an optional label. The label type is created if not existing (maybe after user confirmation) and the label is applied to all the files checked-in. The drawback is that you cannot reuse the comments in the checked-out files.

**Comment by fr43nk** [comment]
HI,
thanks for your request.
As I understand you right, you need to create a label within vscode and configure somehow, that this label is attached on checkin?
Maybe it is then sufficient to have a separate command for label creation? And for the checkin, maybe we need a configuration to activate the checkin+label action?

**Comment by mgrojo** [comment]
Well, the requirement is that all the checked-in files related to an issue are labelled with the same id. That's why I think a good option, that wouldn't interfere with other workflows, is to accept a label as optional input to the check-in command.

**Comment by fr43nk** [comment]
Ok. I will try to integrate such a feature and will provide a prerelease for you to test.

**Comment by mgrojo** [comment]
Thank you for this!

I'm a bit confused, because it seemed to work good in a file, but not in later files that I tried. I don't see any error either. The file is checked-in but the label that I provided to the new prompt is not attached to the file version. How can I investigate this?

**Comment by fr43nk** [comment]
You can check the output console. There you see all the commands that are executed by the extension. How did you check in the first file? Via the SCM view or by the command?

best regards

**Comment by mgrojo** [comment]
I used the Source Control panel.

I will check the output console tomorrow.

**Comment by mgrojo** [comment]
I've made some tests and I haven't seen any clue of what happens with the labelling.

This is what I see in the output console.
```
ci,-comment,Test comment,/snapshot_view/vob/test.txt
Loading "vob/test.txt" (5103 bytes).
Checked in "test.txt" version "/main/project/rel/dev/2".
describe,-fmt,"%m||%Vn",/snapshot_view/vob/test.txt
version||/main/project/rel/dev/2
lsco,-me,-cview,-short,-avobs
```

**Comment by fr43nk** [comment]
Hi,
I tested some use cases and for me it seems to work. Attached you can find a new version. Meanwhile I did some more work on the extension. You can now control the amount of messages shown in the output console. To see all messages just use the configuration: `vscode-clearcase.logLevel: Trace`
If set you should see something like "created label" or "moved label" after entering a name.
Hope this works.

Best regards

**Comment by mgrojo** [comment]
>  If set you should see something like "created label" or "moved label" after entering a name.

And what happens if the label type already exists? Could be that the label is only attached (`mklabel`) after creation (`mklbtype`) works? It could be that I used both cases and only worked one of them.

I will test on Monday with this new version and tell you what the traces say.

**Comment by fr43nk** [comment]
This should work. If a label type already exists then a label is just attached to the version object. And if its already attached to a specific version then it is moved to the checked in version.

**Comment by mgrojo** [comment]
Strange, with [vscode-clearcase-5.0.0-beta.9](https://github.com/user-attachments/files/17678269/vscode-clearcase-5.0.0-beta.9.vsix.zip), I'm not asked for a label. This is the trace output.

```

project.env@@/main/project_nc/rel_nc/dev_nc/ticket_94/CHECKEDOUT from /main/project_nc/rel_nc/dev_nc/ticket_94/0          Rule: CHECKEDOUT

ci,-comment,SYS_12: initial implementation,/data/rational/sources/nc_sys_03_ss/vob/subsys/scripts/project.env
Loading "vob/subsys/scripts/project.env" (2401 bytes).
Checked in "project.env" version "/main/project_nc/rel_nc/dev_nc/ticket_94/1".

ls,/data/rational/sources/nc_sys_03_ss/vob/subsys/scripts/project.env
ls,/data/rational/sources/nc_sys_03_ss/vob/subsys/scripts/project.env
project.env@@/main/project_nc/rel_nc/dev_nc/ticket_94/1      Rule: .../ticket_94/LATEST

project.env@@/main/project_nc/rel_nc/dev_nc/ticket_94/1      Rule: .../ticket_94/LATEST

```

**Comment by mgrojo** [comment]
Sorry, my fault. I had lost the setting for making the label, and it was taking the default (false). It works now after activating that setting!

**Comment by fr43nk** [comment]
Feature is included with #175
